index.js

```diff
@@ -3,40 +3,42 @@
 var $Map = typeof Map === 'function' && Map.prototype ? Map : null;
 var $Set = typeof Set === 'function' && Set.prototype ? Set : null;
 
-if (!$Set) {
-	// eslint-disable-next-line no-unused-vars
-	module.exports = function isSet(x) {
-		// `Set` is not present in this environment.
-		return false;
-	};
-	return;
-}
+(function () {
```

Comment:
Why not just export the full `isSet` function instead of evaluating a function that exports another function? 😉 This way the module looks like a proper module that could eventually be converted to a static ES module and does not depend on any node's quirk and runtime behavior specifics.

Consider this:
```javascript
'use strict';

module.exports = function isSet(x) {
  var $Map = typeof Map === 'function' && Map.prototype ? Map : null;
  var $Set = typeof Set === 'function' && Set.prototype ? Set : null;

  if (!$Set) {
    // `Set` is not present in this environment.
    return false;
  }

  var $mapHas = $Map ? Map.prototype.has : null;
  var $setHas = $Set ? Set.prototype.has : null;

  if (!$setHas) {
    return false;
  }

  if (!x || typeof x !== 'object') {
    return false;
  }

  try {
    $setHas.call(x);

    if ($mapHas) {
      try {
        $mapHas.call(x);
      } catch (e) {
        return true;
      }
    }

    return x instanceof $Set; // core-js workaround, pre-v2.5.0
  } catch (e) {}

  return false;
};
```

Reply:
I just came here for a hotfix, but that's a solid suggestion 😄